npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser.rb

```diff
@@ -233,11 +233,7 @@ def semver_version_for(name, requirement, manifest_name)
           manifest_name: manifest_name
         )&.fetch("version", nil)
 
-        return unless lock_version
-        return if lock_version.include?("://")
-        return if lock_version.include?("file:")
-        return if lock_version.include?("link:")
-        return if lock_version.include?("#")
+        return unless version_class.correct?(lock_version)
```

Comment:
@rebelagentm we should probably add a comment explaining what this excludes and why this is needed, can for-see myself forgetting in future ;)